### ISSUE TITLE
BoxShadow on Border Appears Broken

### ISSUE THREAD

**Issue by robloo** [AvaloniaUI/Avalonia#10539]
**Describe the bug**

BoxShadow on Border Appears Broken

**To Reproduce**

From ColorPreviewer.xaml in Fluent theme.

```xaml
            <Border Grid.Column="1"
                    HorizontalAlignment="Stretch"
                    VerticalAlignment="Stretch"
                    BoxShadow="0 0 10 2 #BF000000"
                    CornerRadius="{TemplateBinding CornerRadius}"
                    Margin="10">
              <Panel>
                <Border Background="{StaticResource ColorControlCheckeredBackgroundBrush}"
                        CornerRadius="{TemplateBinding CornerRadius}" />
                <Border CornerRadius="{TemplateBinding CornerRadius}"
                        Background="{TemplateBinding HsvColor, Converter={StaticResource ToBrushConverter}}"
                        HorizontalAlignment="Stretch"
                        VerticalAlignment="Stretch" />
              </Panel>
            </Border>
```

**Expected behavior**

**Screenshots**

![image](https://user-images.githubusercontent.com/17993847/222624923-a095c25b-11e5-40c1-92e4-59d89a75edb1.png)

**Desktop (please complete the following information):**

- OS: [e.g. Windows, Mac, Linux (State distribution)]
- Version 92562b0d9979becc8d8c2316e1c2f0678bdae19d

**Additional context**

**Comment by rabbitism** [comment]
As a reference, what it looks like before. 
![image](https://user-images.githubusercontent.com/14807942/222638585-7538c9c1-3ab8-40d4-bf6e-35e6e76a0382.png)

**Comment by laolarou726** [comment]
Try give Border a background

**Comment by timunie** [comment]
Yes @laolarou726, seems like you are correct. Setting the Background to `Transparent` works, but the default `x:Null` does not. 

```xml
<Border Background="Transparent"   <-- This is the important line
        Grid.Column="1"
        HorizontalAlignment="Stretch"
        VerticalAlignment="Stretch"
        BoxShadow="0 0 10 2 #BF000000"
        CornerRadius="{TemplateBinding CornerRadius}"
        Margin="10">
  <Panel>
    <Border Background="{StaticResource ColorControlCheckeredBackgroundBrush}"
            CornerRadius="{TemplateBinding CornerRadius}" />
    <Border CornerRadius="{TemplateBinding CornerRadius}"
            Background="{TemplateBinding HsvColor, Converter={StaticResource ToBrushConverter}}"
            HorizontalAlignment="Stretch"
            VerticalAlignment="Stretch" />
  </Panel>
</Border>

```

![image](https://user-images.githubusercontent.com/47110241/224260761-d2b33a25-3c7b-49ab-878d-cf9db3449693.png)

@kekekeks I think it's by design, that `null` does not render a shadow?

**Comment by Gillibald** [comment]
The compositor might ignore drawing operations with a null brush. Just some theory haven't checked the actual code.

**Comment by robloo** [comment]
Yea, that makes sense. It's just strange it changed at some point since the last preview. So wanted to confirm it's correct and intended.

**Comment by kekekeks** [comment]
It's probably this check - https://github.com/AvaloniaUI/Avalonia/blob/92562b0d9979becc8d8c2316e1c2f0678bdae19d/src/Avalonia.Base/Media/DrawingContext.cs#L135-L136

**Comment by robloo** [comment]
I'm going to use the Transparent background work-around in the ColorPicker. Again, please note that the drop shadow requiring a background to appear is changed from earlier previews. Before it would show even with an unset Background.

If this new behavior is deemed the correct functionality (and it is somewhat reasonable) this issue can be closed. That will simply be the new functionality going forward.

**Comment by timunie** [comment]
I personally think this is intended. Therefore I'm closing this issue as not planned. If someone disagrees, let me know.

**Comment by robloo** [comment]
@timunie I don't disagree, someone had to make the call. It might be worth it to list as a breaking change though.

**Comment by kekekeks** [comment]
It's a bug, not a behavior change

**Comment by wpfblendperson** [comment]
Please put opacity on dropshadow! very important!

**Comment by timunie** [comment]
@wpfblendperson this issue is marked as resolved. So should be available in 11.1 beta latest. 

In general, don't answer closed issues. If you need priority on issues, you can request this via paid support.

**Comment by robloo** [comment]
This was resolved for 11.0 itself. Please open a separate feature request for unrelated topics.
